src/modules/ce/components/ClientCeEligibility.tsx

```diff
@@ -0,0 +1,134 @@
+import { Stack, Typography } from '@mui/material';
+import { useCallback, useContext, useState } from 'react';
+import LoadingButton from '@/components/elements/LoadingButton';
+import AssessmentContext from '@/modules/assessments/components/AssessmentContext';
+import ApolloErrorAlert from '@/modules/errors/components/ApolloErrorAlert';
+import ErrorAlert from '@/modules/errors/components/ErrorAlert';
+import { emptyErrorState, ErrorState, hasErrors } from '@/modules/errors/util';
+import { FormDefinitionHandlers } from '@/modules/form/hooks/useFormDefinitionHandlers';
+import { DynamicInputCommonProps } from '@/modules/form/types';
+import { localResolvePickList } from '@/modules/form/util/formUtil';
+import {
+  CalculateClientCeEligibilityMutation,
+  PickListOption,
+  useCalculateClientCeEligibilityMutation,
+} from '@/types/gqlTypes';
+
+const projectTypePickList = localResolvePickList('ProjectType');
+
+interface ClientEligibilityProps extends DynamicInputCommonProps {
+  value?: PickListOption[] | null;
+  onChange?: (value: PickListOption[] | null) => void;
+  handlers?: FormDefinitionHandlers;
+}
+
+// Shows a button to calculate client eligibility based on current form values, and displays the result.
+const ClientCeEligibility = ({
+  value,
+  onChange,
+  label,
+  disabled = false,
+  handlers,
+}: ClientEligibilityProps) => {
+  const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
+
+  const [calculateClientEligibility, { loading }] =
+    useCalculateClientCeEligibilityMutation({
+      onCompleted: (data: CalculateClientCeEligibilityMutation) => {
+        const errors = data.calculateClientCeEligibility?.errors || [];
+        if (errors.length > 0) {
+          setErrorState({ ...emptyErrorState, errors });
+          return;
+        }
+
+        setErrorState(emptyErrorState);
+
+        if (onChange && data.calculateClientCeEligibility?.projectTypes) {
+          const eligibleProjectTypes =
+            data.calculateClientCeEligibility.projectTypes;
+
+          // Convert the returned project type codes to their corresponding pick list options.
+          // Filter out undefined values to keep typescript happy
+          const matchingPickListOptions = eligibleProjectTypes
+            .map((projectTypeCode) =>
+              projectTypePickList?.find(
+                (pickListOption) => pickListOption.code === projectTypeCode
+              )
+            )
+            .filter((option): option is PickListOption => option !== undefined);
+
+          onChange(matchingPickListOptions);
```

Comment:
Suggestion to make this a bit more concise, without changing the logic. It would be nice to have a comment explaining why this needs to be represented as an array of PickListOptions (as opposed to an array of project types) but I wasn't totally clear on that. I think it's probably because we have some other code expecting that array values are always pick list shaped?

```suggestion
          // Convert the returned project type codes to pick list options
          // (because..? this is this expected format for form items that collect multi-value arrays?)
          const eligibleProjectTypesAsPickList = eligibleProjectTypes.map(
            (code) => ({ code, label: HmisEnums.ProjectType[code] })
          );
          onChange(eligibleProjectTypesAsPickList);
```

Reply:
Yes that's right, that code is here: https://github.com/greenriver/hmis-frontend/blob/6c27c774eb1ed1b78f09e79f0fb7a888a2757322/src/modules/form/util/formUtil.ts#L799

I've added a comment referring to that method

Reply:
I see, thanks!